resources/js/beatmap-discussions/editor-helpers.ts

```diff
@@ -77,38 +77,39 @@ function serializeEmbed(node: EmbedElement): DocumentIssueEmbed {
   }
 }
 
-function serializeParagraph(node: ParagraphElement) {
-  const childOutput: string[] = [];
-  const currentMarks = {
-    bold: false,
-    italic: false,
-  };
-
-  node.children.forEach((child) => {
-    if (child.text !== '') {
-      if (currentMarks.bold !== (child.bold ?? false)) {
-        currentMarks.bold = (child.bold as boolean) ?? false;
-        childOutput.push('**');
-      }
-
-      if (currentMarks.italic !== (child.italic ?? false)) {
-        currentMarks.italic = (child.italic as boolean) ?? false;
-        childOutput.push('*');
-      }
-    }
+// Prevent invalid markdown from being generated by whitespace after/before the opening/closing marks
+function serializeMarkedText(text: string, format: string) {
+  const trimmedText = text.trim();
+  if (trimmedText.length === 0) {
+    return;
+  }
 
-    childOutput.push((child.text).replace('*', '\\*'));
-  });
+  const formattedText = `${format}${trimmedText}${format}`;
 
-  // ensure closing of open tags
-  if (currentMarks.bold) {
-    childOutput.push('**');
-  }
-  if (currentMarks.italic) {
-    childOutput.push('*');
+  if (trimmedText === text) {
+    return formattedText;
   }
 
-  return childOutput.join('');
+  return text.replace(trimmedText, formattedText);
+}
+
+function serializeParagraph(node: ParagraphElement) {
+  return node.children.map((child) => {
+    if (child.text !== '') {
+      const text = child.text.replace(/\*/g, '\\*');
```

Comment:
should probably at least escape `\` as well (although at this point this looks like a losing battle esp if we're to support more formatting)